src/utils/editor/expression.js

```diff
@@ -46,7 +46,25 @@ export function updatePreview(
 
   const isUpdating = preview && preview.updating;
 
-  const inScope = linesInScope && linesInScope.includes(location.line);
+  let inScope;
+  const isLineInScope = linesInScope && linesInScope.includes(location.line);
+  if (isLineInScope) {
+    inScope = true;
+  } else {
+    const startLineLocation = outOfScopeLocations.find(
+      loc => loc.start.line === location.line
+    );
+    if (startLineLocation) {
```

Comment:
we seem to be using the out of scope locations to determine if the lien is in scope, but we have a linesInScope array -- why was this not working as intended? perhaps the bug is steming from linesInScope not representing the correct information?

Reply:
yes, actually `linesInScope` is not really mandatory here, since we need the column information.
you could write a function which works only with `outOfScopeLocations` and calculates if a location is out of scope.
I kept `linesInScope` since it was easier to calculate only in for cases where the line isn't in that array.

hope that makes sense o_O